### PR TITLE
Use OwnedTableReference for subquery aliases

### LINKED ISSUE #6016: Unable to query view with a qualified name
### Describe the bug

Creating a view in a new schema then trying to querying that view using its qualified name results in an error. See below:

```
DataFusion CLI v22.0.0
❯ create schema s;
0 rows in set. Query took 0.007 seconds.
❯ create view s.v as (select 1 as a);
0 rows in set. Query took 0.004 seconds.
❯ select * from s.v;
Schema error: No field named s.v.a. Valid fields are "s.v".a.
```

I'm also not able to use fully qualified field names to select from this view either:

```
❯ select a from s.v;
Schema error: No field named s.v.a. Valid fields are "s.v".a.
❯ select s.v.a;
Schema error: No field named s.v.a.
❯ select "s.v".a;
Schema error: No field named s.v.a.
```

### To Reproduce

Using datafusion-cli

```
DataFusion CLI v22.0.0
❯ create schema s;
0 rows in set. Query took 0.007 seconds.
❯ create view s.v as (select 1 as a);
0 rows in set. Query took 0.004 seconds.
❯ select * from s.v;
Schema error: No field named s.v.a. Valid fields are "s.v".a.
```

### Expected behavior

I can run `select * from s.v` (along with the qualified alternatives).

### Additional context

Also fails in DataFusion 21.1.0 & 21.0.0:

```
DataFusion CLI v21.1.0
❯ create schema s;
0 rows in set. Query took 0.007 seconds.
❯ create view s.v as select 1 as a;
0 rows in set. Query took 0.004 seconds.
❯ select * from s.v
;
Schema error: No field named "s"."v"."a". Valid fields are "s.v"."a".
```

Works as expected in 20.0.0

```
DataFusion CLI v20.0.0
❯ create schema s;
0 rows in set. Query took 0.008 seconds.
❯ create view s.v as select 1 as a;
0 rows in set. Query took 0.004 seconds.
❯ select * from s.v;
+---+
| a |
+---+
| 1 |
+---+
1 row in set. Query took 0.010 seconds.
```

### PR DESCRIPTION
**Pull request by scsmithr**
# Which issue does this PR close?

Closes #6016.

(Opening this to get early feedback)

# Rationale for this change

Simplest way I found for fixing the above issue. I didn't want to try to add an analyzer rule for swapping out references in parent nodes.

# What changes are included in this PR?

Changes `SubqueryAlias` to hold a `OwnedTableReference` instead of just a string.

# Are these changes tested?

Adds sqllogictests.

# Are there any user-facing changes?

Yes, this changes the public interface for `SubqueryAlias`.

### CODE REVIEW

**Comment by Jefffrey** [comment]
I did find some previous discussion regarding this for what it's worth: https://github.com/apache/arrow-datafusion/pull/5615#discussion_r1137647079 @scsmithr 

cc @alamb

File: datafusion/expr/src/logical_plan/builder.rs [hunk 1]
```diff
@@ -381,7 +381,7 @@ impl LogicalPlanBuilder {
     }
 
     /// Apply an alias
-    pub fn alias(self, alias: impl Into<String>) -> Result<Self> {
+    pub fn alias(self, alias: impl Into<OwnedTableReference>) -> Result<Self> {
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/builder.rs:384]: 👍  this change makes a lot of sense to me and is consistent with creating names of other relations (like `CreateExternalTable`)

https://docs.rs/datafusion-expr/22.0.0/datafusion_expr/logical_plan/struct.CreateExternalTable.html

File: datafusion/expr/src/logical_plan/plan.rs [hunk 2]
```diff
@@ -1304,20 +1304,20 @@ pub struct SubqueryAlias {
...
 }
 
 impl SubqueryAlias {
-    pub fn try_new(plan: LogicalPlan, alias: impl Into<String>) -> Result<Self> {
+    pub fn try_new(
+        plan: LogicalPlan,
+        alias: impl Into<OwnedTableReference>,
+    ) -> Result<Self> {
         let alias = alias.into();
-        let table_ref = TableReference::bare(&alias);
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:1315]: 👍

**Review by alamb — COMMENTED** [review]
Thank you @scsmithr  and @Jefffrey  -- this looks change looks good to me 👍

**Comment by scsmithr** [comment]
CI errors unrelated, see https://github.com/apache/arrow-datafusion/pull/6028

Otherwise I think this is good to go.

**Review by yjshen — APPROVED** [review]

**Review by r4ntix — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @scsmithr

**Pull request merged**